common/src/Assets/EntityDefinition.h

```diff
@@ -59,7 +60,7 @@ namespace TrenchBroom {
             std::string m_name;
             Color m_color;
             std::string m_description;
-            size_t m_usageCount;
+            std::unique_ptr<std::atomic<size_t>> m_usageCount;
```

Comment:
Why do you use a unique pointer here?

Reply:
std::atomic is not movable, but this class needs to be, so I wrapped it in a unique_ptr

Reply:
I don't think `EntityDefinition` and its subclasses don't need to be moveable. In fact, they probably aren't because `EntityDefinition` has a custom destructor.

But I do see the point for `Texture`. I would consider manually implementing the move constructor / assignment for `Texture´ by explicitly copying the usage count, though. The unique_ptr is looks really odd to me.

Reply:
I got compiler errors saying that, I'm pretty sure on both classes, but I can double check..

Reply:
This is just a hack, the atomics should be removed later anyway, so I don't see the problem if it looks odd. If it's just aesthetics I don't want to make further changes here. Unless the way I'm using unique_ptr is wrong somehow?

Reply:
Thanks for changing it. It's not just about aesthetics, it's about communicating intent. When I see this, I wonder why it's done this way and I would never have guessed the reason.